plugins/inputs/ravendb/ravendb.go

```diff
@@ -0,0 +1,680 @@
+	MappedPerSec               float64  `json:"MappedPerSec"`
+	ReducedPerSec              float64  `json:"ReducedPerSec"`
+	Type                       string   `json:"Type"`
+	EntriesCount               int32    `json:"EntriesCount"`
+}
+
+type collectionsMetricResponse struct {
+	Results         []*collectionMetrics `json:"Results"`
+	PublicServerUrl string               `json:"PublicServerUrl"`
+	NodeTag         string               `json:"NodeTag"`
+}
+
+type collectionMetrics struct {
+	DatabaseName          string `json:"DatabaseName"`
+	CollectionName        string `json:"CollectionName"`
+	DocumentsCount        int64  `json:"DocumentsCount"`
+	TotalSizeInBytes      int64  `json:"TotalSizeInBytes"`
+	DocumentsSizeInBytes  int64  `json:"DocumentsSizeInBytes"`
+	TombstonesSizeInBytes int64  `json:"TombstonesSizeInBytes"`
+	RevisionsSizeInBytes  int64  `json:"RevisionsSizeInBytes"`
+}
+
+var sampleConfig = `
+  ## Node URL and port that RavenDB is listening on.
+  url = "https://localhost:8080"
+
+  ## RavenDB X509 client certificate setup
+  tls_cert = "/etc/telegraf/raven.crt"
+  tls_key = "/etc/telegraf/raven.key"
+
+  ## Optional request timeouts
+  ##
+  ## ResponseHeaderTimeout, if non-zero, specifies the amount of time to wait
+  ## for a server's response headers after fully writing the request.
+  # header_timeout = "3s"
+  ##
+  ## client_timeout specifies a time limit for requests made by this client.
+  ## Includes connection time, any redirects, and reading the response body.
+  # client_timeout = "4s"
+
+  ## When true, collect server stats
+  # gather_server_stats = true
+
+  ## When true, collect per database stats
+  # gather_db_stats = true
+
+  ## When true, collect per index stats
+  # gather_index_stats = true
+
+  ## When true, collect per collection stats
+  # gather_collection_stats = true
+
+  ## List of db where database stats are collected
+  ## If empty, all db are concerned
+  # db_stats_dbs = []
+
+  ## List of db where index status are collected
+  ## If empty, all indexes from all db are concerned
+  # index_stats_dbs = []
+
+  ## List of db where collection status are collected
+  ## If empty, all collections from all db are concerned
+  # collection_stats_dbs = []
+`
+
+// SampleConfig ...
+func (r *RavenDB) SampleConfig() string {
+	return sampleConfig
+}
+
+// Description ...
+func (r *RavenDB) Description() string {
+	return "Reads metrics from RavenDB servers via the Monitoring Endpoints"
+}
+
+// Gather ...
+func (r *RavenDB) Gather(acc telegraf.Accumulator) error {
+	err := r.ensureClient()
+	if nil != err {
+		if nil != r.Log {
+			r.Log.Errorf("Error with Client %s", err)
+		}
+		return err
+	}
+
+	var wg sync.WaitGroup
+	wg.Add(len(gatherFunctions))
+	for _, f := range gatherFunctions {
+		go func(gf gatherFunc) {
+			defer wg.Done()
+			gf(r, acc)
+		}(f)
+	}
+	wg.Wait()
+
+	return nil
+}
+
+func (r *RavenDB) ensureClient() error {
+	if r.Client == nil {
```

Comment:
Please invert the logic here and save some indention level below:
```suggestion
	if r.Client != nil {
		return nil
	}
```
_or_ move the if up one level.

Reply:
done

Reply:
Please mark as resolved.